src/main/java/se/yolean/kafka/keyvalue/ConsumerAtLeastOnce.java

```diff
@@ -260,7 +285,17 @@ void onNullKey(UpdateRecord update) {
 
   @Override
   public Long getCurrentOffset(String topicName, int partition) {
-    return currentOffsets.get(new TopicPartition(topicName, partition));
+    return currentOffsets.get(new TopicPartition(topicName, partition)).get();
+  }
+
+  @Override
+  public List<Map<String, ?>> getCurrentOffsets() {
```

Comment:
Since we depend on a specific structure of this header, I would create a class for this structure intead of using a generic map so that we get compile time errors instead of relying on the correctness of a test and not seeing errors until that test is run.